gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/engine/access/operations/SortedMap2IndexInsertOperation.java

```diff
@@ -283,6 +284,28 @@ else if (lockTimeout > 0) { // wait before throwing conflict
           }
           break;
         }
+
+        // TODO: identify the cause of Bug SNAP-2627
+        // Till then this is the crude fix
+        if (oldValue instanceof AbstractRegionEntry &&
+            ((AbstractRegionEntry)oldValue).isDestroyedOrRemoved()) {
+          // create a dummy exception
+
+          Throwable th = GemFireXDUtils.newDuplicateKeyViolation("unique constraint",
+              container.getQualifiedTableName(), "key=" + key.toString()
+                  + ", row=" + value, oldValue, null, null);
+
+          final GemFireCacheImpl cache = Misc.getGemFireCache();
+          final LogWriter logger = cache.getLogger();
+
+          logger.error("Unique index constraint violation caused due to " +
+              "removed/destroyed entry. The index is corrupted. Cleaning the index", th);
+
```

Comment:
You want to remove all unnecessary blank spaces. @suranjan and @sumwale please look into this small PR.